### DISCUSSION TITLE
Adding Entity Ruler to Custom NER Model

### DISCUSSION THREAD

**Discussion by morankyle** [explosion/spaCy#8414] [Help: Coding & Implementations]
Struggling to correctly add this entity ruler to my current Spacy v3 model pipeline, tried using an entry point but that didn't seem to work. Is there an example of just adding the component to the piepline via the model's config file?

```
import spacy
from spacy.language import Language
from spacy.tokens import Span

nlp = spacy.load("en_core_web_sm")

@Language.component("expand_person_entities")
def expand_person_entities(doc):
    new_ents = []
    for ent in doc.ents:
        if ent.label_ == "PERSON" and ent.start != 0:
            prev_token = doc[ent.start - 1]
            if prev_token.text in ("Dr", "Dr.", "Mr", "Mr.", "Ms", "Ms."):
                new_ent = Span(doc, ent.start - 1, ent.end, label=ent.label)
                new_ents.append(new_ent)
        else:
            new_ents.append(ent)
    doc.ents = new_ents
    return doc

# Add the component after the named entity recognizer
nlp.add_pipe("expand_person_entities", after="ner")
```

**Comment by morankyle** [comment]
I poorly worded this question but I'm getting an error that the custom component isn't registered with the language class, but shouldn't that code above register it with the language class?

**Comment by polm** [comment]
Based on your comment in https://github.com/explosion/spaCy/discussions/8088#discussioncomment-879030 I think you figured this out, but to have the component registered you need to import the code that defines it (or have it declared above where you use the component).

**Reply by polm** [reply]
Also note that if you're training with a config file, you can pass `-c myfile.py` to load a file with registered functions you need for your pipeline. See [the docs](https://spacy.io/api/cli#train) for reference.
